multipart-basic/src/utils/payload_handler.rs

```diff
@@ -0,0 +1,99 @@
+use actix_multipart::{Field, Multipart};
+use actix_web::{web};
+use futures::StreamExt;
+use serde::{Deserialize, Serialize};
+use std::io::Write;
+use std::{str};
+
+
+
+#[derive(Debug, Clone)]
+pub struct UploadedFiles {
+    pub name: String,
+    pub path: String,
+}
+impl UploadedFiles {
+    fn new(filename: &str) -> UploadedFiles {
+        UploadedFiles {
+            name: filename.to_string(),
+            path: format!("./files/{}", filename),
+        }
+    }
+
+}
+
+#[derive(Deserialize, Serialize, Debug)]
+pub struct FormOut {
+    title: String,
+    description: String, 
+    count: u32
+}
+
+pub async fn split_payload(payload: &mut Multipart) -> (FormOut, Vec<UploadedFiles>) {
+    let mut files: Vec<UploadedFiles> = Vec::new();
+
+    /* fill with default values for now */
+    let mut form : FormOut=FormOut {
+        title: "".to_string(),
+        description: "".to_string(),
+        count: 0
+    };
+
+    while let Some(item) = payload.next().await {
+        let mut field: Field = item.expect(" split_payload err");
+        let content_type = field.content_disposition().unwrap();
+        let name = content_type.get_name().unwrap();
+        if name != "file" {
+            while let Some(chunk) = field.next().await {
+                let data = chunk.expect("split_payload err chunk");
+                /* convert bytes to string and print it  (just for testing) */
+                if let Ok(s)=str::from_utf8(&data){
+                    println!("{:?}", s);

```

Comment:
if we're going to include debug prints, can we use log + env_logger and use debug! macro here